ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -0,0 +1,1072 @@
+            return line.getProperties().getProperty(UcteImporter.GEOGRAPHICAL_NAME_PROPERTY_KEY);
+        }
+        return "";
+    }
+
+    UctePowerPlantType energySourceToUctePowerPlantType(EnergySource energySource) {
+        if (EnergySource.HYDRO == energySource) {
+            return UctePowerPlantType.H;
+        } else if (EnergySource.NUCLEAR == energySource) {
+            return UctePowerPlantType.N;
+        } else if (EnergySource.THERMAL == energySource) {
+            return UctePowerPlantType.C;
+        } else if (EnergySource.WIND == energySource) {
+            return UctePowerPlantType.W;
+        } else {
+            return UctePowerPlantType.F;
+        }
+    }
+
+    /**
+     * "increment" the string given in parameter by 1 : "aaaa" gives "aaab", "aaaz" gives "aaaA" etc...
+     * It goes from a -> z -> A -> Z -> 0 -> 9 and it's used to generate a random geographicalName.
+     * @param toIncrement the string to increment
+     * @return a string "incremented" by 1
+     */
+    String incrementGeneratedGeographicalName(String toIncrement) {
+        if (toIncrement.length() < 4) {
+            throw new IllegalArgumentException("The string to increment is not long enough (should be at least 4)");
+        }
+        char charAt3 = toIncrement.charAt(3);
+        char charAt2 = toIncrement.charAt(2);
+        char charAt1 = toIncrement.charAt(1);
+        char charAt0 = toIncrement.charAt(0);
+
+        if (POSSIBLE_CHARACTER.indexOf(charAt3) == POSSIBLE_CHARACTER.size() - 1) {
+            if (POSSIBLE_CHARACTER.indexOf(charAt2) == POSSIBLE_CHARACTER.size() - 1) {
+                if (POSSIBLE_CHARACTER.indexOf(charAt1) == POSSIBLE_CHARACTER.size() - 1) {
+                    if (POSSIBLE_CHARACTER.indexOf(charAt0) == POSSIBLE_CHARACTER.size() - 1) {
+                        return "aaaa";
+                    } else {
+                        charAt3 = (char) POSSIBLE_CHARACTER.get(0);
+                        charAt2 = (char) POSSIBLE_CHARACTER.get(0);
+                        charAt1 = (char) POSSIBLE_CHARACTER.get(0);
+                        charAt0 = (char) POSSIBLE_CHARACTER.get(POSSIBLE_CHARACTER.indexOf(charAt0) + 1);
+                    }
+                } else {
+                    charAt3 = (char) POSSIBLE_CHARACTER.get(0);
+                    charAt2 = (char) POSSIBLE_CHARACTER.get(0);
+                    charAt1 = (char) POSSIBLE_CHARACTER.get(POSSIBLE_CHARACTER.indexOf(charAt1) + 1);
+                }
+            } else {
+                charAt3 = (char) POSSIBLE_CHARACTER.get(0);
+                charAt2 = (char) POSSIBLE_CHARACTER.get(POSSIBLE_CHARACTER.indexOf(charAt2) + 1);
+            }
+        } else {
+            charAt3 = (char) POSSIBLE_CHARACTER.get(POSSIBLE_CHARACTER.indexOf(charAt3) + 1);
+        }
+        return String.valueOf(charAt0) + String.valueOf(charAt1) + String.valueOf(charAt2) + String.valueOf(charAt3);
+    }
+}
```

Comment:
I really don't understand what you are trying to do... I'm not sure you need this function. Generating IDs should be done using a naming strategy. This strategy is not the good one to me.

Reply:
We agreed for a first version to generate the missing informations this way but we indeed need to find a way to generate them in a better way. Any proposition ?